Index hero

lilyfielding opened the pull request:
## 13th April 2020

Hey Laura! Here's everything that went down yesterday:

### Notes
- Added schema, liquid code and responsive styling.
- Since I don't have the file for the background image yet, I've added a grey background to the section for now.
- Created `_typography.scss` for type-related helper classes and H1 styles
- Added flexbox helper classes to `_helpers.scss`
- Added helper class for full-height sections.
- Created helper classes for `.button` and `.button-white`, to apply to the bordered buttons on the site. This meant removing the class of `.button` from the search button, as this doesn't share the same styling as the others.
- Added `reset-input` mixin to make styling the inputs easier.

### Issues
- When you rename a file it gets pushed to Shopify as a completely new file, and the original one with the old name doesn't get deleted. This can be overcome by going into the Shopify code editor and deleting the old file manually, but you'll have to go back and re-add any content added via the theme customiser.
- The header isn't responsive yet so this is causing some jankiness on smaller screen sizes.

1aurabrown commented:
Thanks, will talk a look soon!

The issues of files not getting deleted is something that can be fixed in webpack or the build scripts. I will check it out.

1aurabrown commented on src/styles/global/_typography.scss:
```diff
@@ -0,0 +1,29 @@
+
+/*============= Elements ================*/
+
+h1 {
```
could we assign this to a class name like .h1? it is likely that there will be instances where the structural h-level will not directly map to type hierarchy.

i also noticed that we should set the font-weight on all header elements to normal—the browser is automatically applying a bold style but actually all the type is supposed to be normal weight.

lilyfielding replied:
Yep! I will change the markup on this. If you want, I can add a stylesheet to reset all of the default browser styles?

1aurabrown replied:
:)

1aurabrown commented on src/styles/util/_helpers.scss:
```diff
@@ -23,6 +23,32 @@
+.center-both {
+  justify-content: center;
+  align-items: center;
+}
```
actually some styles like this already exist via a library i installed, svbstrate. see below:
```css
.f { display: flex }
.fw { flex-wrap: wrap }
.ais { align-items: flex-start }
.aie { align-items: flex-end }
.aic { align-items: center }
.aib { align-items: baseline }
.jcs { justify-content: flex-start }
.jce { justify-content: flex-end }
.jcc { justify-content: center }
.jcb { justify-content: space-between }
.fa {
  flex: 1 1 auto;
  min-width: 0;
  min-height: 0;
}

```

so you can add combinations of these classes to html elements to achieve the desired flexbox effect

more helpers included with the library can be found here, sorry for not mentioning it sooner!: https://github.com/estrattonbailey/svbstrate/tree/master/src/lib

lilyfielding replied:
My bad, I should have checked for this. I'll remove the helpers I wrote and replace the class names with these shorter ones

1aurabrown commented on src/sections/hero.liquid:
```diff
@@ -0,0 +1,62 @@
+{% assign img_url = settings.background_image | img_url: 'master' %}
+
+<section class="section">
+  <div class="hero flex-column center-both full-height" style="{% if settings.background_image %} background-image: url({{ img_url}}; {% endif %}">
```
not sure this necessarily has to be full height, but fine for now—can change it later. depending on the client's content we might want to allow a separate mobile image upload.

1aurabrown commented on src/sections/hero.liquid:
```diff
@@ -0,0 +1,62 @@
+{% assign settings = section.settings %}
+{% assign img_url = settings.background_image | img_url: 'master' %}
```
it would be great if we could use responsive images. if i haven't already, i will install [this library](http://afarkas.github.io/lazysizes/rias/) for this purpose, but it requires using an img element as opposed to a background image. I can abstract the image tag into a snippet later.

lilyfielding replied:
Ooh thanks for this, I've never seen this library before! I'll change the image to a tag so we can set this up later

lilyfielding commented:
Just made some quick edits:
1. Changed `h1` styles to `.h1`
2. Deleted some of the custom helper classes and changed the names to svbstrate ones
3. Changed the background image to a tag

1aurabrown commented:
Great thank you!
